tests/run.sh

```diff
@@ -63,18 +96,50 @@ EOF
 start_services() {
     stop_services
     clean_data
+    generate_tls_keys
+
+    cat - > "$OUT_DIR/pd-config.toml" <<EOF
+[security]
+# set the path for certificates. Empty string means disabling secure connectoins.
+cacert-path = "$OUT_DIR/cert/ca.pem"
+cert-path = "$OUT_DIR/cert/pd.pem"
+key-path = "$OUT_DIR/cert/pd.key"
+EOF
 
     echo "Starting PD..."
     pd-server \
-        --client-urls http://127.0.0.1:2379 \
+        --client-urls https://127.0.0.1:2379 \
+        --peer-urls https://127.0.0.1:2380 \
+        --advertise-client-urls https://127.0.0.1:2379 \
+        --advertise-peer-urls https://127.0.0.1:2380 \
+        -config "$OUT_DIR/pd-config.toml" \
         --log-file "$OUT_DIR/pd.log" \
         --data-dir "$OUT_DIR/pd" &
 
     # wait until PD is online...
-    while ! curl -o /dev/null -sf http://127.0.0.1:2379/pd/api/v1/version; do
+    while ! wget -q -O - \
+    --ca-certificate="$OUT_DIR/cert/ca.pem" \
+    --certificate="$OUT_DIR/cert/client.pem" \
+    --private-key="$OUT_DIR/cert/client.key" \
+    https://127.0.0.1:2379/pd/api/v1/version; do
         sleep 1
     done
 
+
+    # on CI curl's version is too old: curl: (58) unable to load client key: -8178 (SEC_ERROR_BAD_KEY)))
```

Comment:
can we remove it now?

Reply:
removed https://github.com/pingcap/tidb-binlog/pull/904/commits/dfe4e41bf6639ddebbe279024cb34ce858097f61